File: FRONTEND/src/types/me.ts
import type { Course } from './course';

export type PurchasedCourse = {
  venta_id: number;
  course_id: number;
  name: string | null;
  detalle: string | null;
  image: string | null;
  progreso: number;
  cantidad_v: number;
};

export type VideoProgress = {
  id: number;
  title: string;
  content: string;
  seccion: number;
  title_accordion: string | null;
  url: string | null;
  file: string | null;
  descarga: string | null;
  completed: boolean;
};

export type CourseContent = {
  course: Course;
  videos: VideoProgress[];
  progreso: number;
  cantidad_v: number;
  total_videos: number;
};

File: BACKEND/app/schemas/me.py
from pydantic import BaseModel, Field

from app.schemas.course import CoursePublic


class PurchasedCoursePublic(BaseModel):
    venta_id: int
    course_id: int
    name: str | None = None
    detalle: str | None = None
    image: str | None = None
    progreso: int = Field(default=0, ge=0, le=100)
    cantidad_v: int = Field(default=0, ge=0)


class VideoProgressPublic(BaseModel):
    id: int
    title: str
    content: str
    seccion: int
    title_accordion: str | None = None
    url: str | None = None
    file: str | None = None
    descarga: str | None = None
    completed: bool = False


class CourseContentPublic(BaseModel):
    course: CoursePublic
    videos: list[VideoProgressPublic]
    progreso: int = Field(default=0, ge=0, le=100)
    cantidad_v: int = Field(default=0, ge=0)
    total_videos: int = Field(default=0, ge=0)
